crates/miden-testing/src/kernel_tests/tx/test_account.rs

```diff
@@ -1265,6 +1266,241 @@ fn incrementing_nonce_twice_fails() -> anyhow::Result<()> {
+            assert_eqw.err=\"initial map value should remain unchanged\"
+
+            # get_map_item_init for the new key should return empty word (default)
+            push.{new_key}
+            push.0
+            call.mock_account::get_map_item_init
+            push.0.0.0.0
+            assert_eqw.err=\"new key should have empty initial value\"
+
+            dropw dropw
+        end
+        ",
+        initial_key = &initial_key,
+        initial_value = &initial_value,
+        new_key = &new_key,
+        new_value = &new_value,
+    );
+
+    tx_context.execute_code(&code).unwrap();
+
+    Ok(())
+}
+
+#[test]
+fn test_get_item_init_and_get_map_item_init_with_foreign_account() -> miette::Result<()> {
+    let mut mock_chain = MockChain::builder();
+
+    // Create a native account and add it to the mock chain
+    let native_account = mock_chain
+        .add_existing_mock_account(Auth::IncrNonce)
+        .expect("failed to add native account");
+
+    mock_chain.build().expect("failed to build mock chain");
+
+    let (map_key, map_value) = STORAGE_LEAVES_2[0];
+
+    // Create foreign procedures that test get_item_init and get_map_item_init
+    let foreign_account_code_source = "
+        use.miden::account
+        use.std::sys
+
+
+        export.test_get_item_init
+            push.0
+            exec.account::get_item_init
+            exec.sys::truncate_stack
+        end
+
+        export.test_get_map_item_init
+            exec.account::get_map_item_init
+            exec.sys::truncate_stack
+        end
+    ";
+
+    let foreign_account_component = AccountComponent::compile(
+        foreign_account_code_source,
+        TransactionKernel::with_kernel_library(Arc::new(DefaultSourceManager::default())),
+        vec![AccountStorage::mock_item_0().slot, AccountStorage::mock_item_2().slot],
+    )
+    .unwrap()
+    .with_supports_all_types();
+
+    // Update the foreign account to use the new component
+    let foreign_account = AccountBuilder::new(ChaCha20Rng::from_os_rng().random())
+        .with_auth_component(Auth::IncrNonce)
+        .with_component(foreign_account_component.clone())
+        .build_existing()
+        .unwrap();
+
+    // Rebuild the mock chain with the updated foreign account
+    let mut mock_chain =
+        MockChainBuilder::with_accounts([native_account.clone(), foreign_account.clone()])
+            .unwrap()
+            .build()
+            .unwrap();
+    mock_chain.prove_next_block().unwrap();
+
+    let foreign_account_inputs = mock_chain
+        .get_foreign_account_inputs(foreign_account.id())
+        .expect("failed to get foreign account inputs");
+
+    let code = format!(
+        "
+        use.std::sys
+        use.miden::tx
+
+        begin
+
+            # Test get_item_init on foreign account
+            padw padw padw push.0.0
+            push.{test_get_item_init_hash}
+            push.{foreign_account_id_suffix}.{foreign_account_id_prefix}
```

Comment:
Nit: I would setup foreign account tests using `procref`, I think it's much nicer and less low-level than fetching the raw MAST root, e.g. like here: https://github.com/0xMiden/miden-base/blob/fdf7c0e47191277a95f0ce29abb7b2d7f0a9d754/crates/miden-testing/src/kernel_tests/tx/test_fpi.rs#L574

Side note: I would do the same in miden-tutorials - I think it's better to teach people to link against the foreign account directly at assembly time 🙂